### PR TITLE
cookie baking

### PR DESCRIPTION
**Pull request by danielgustafsson**
This is a small patchset of refactorings and small cleanups of the cookie code. I have a lot more planned but wanted to get some of the diffs out there in a more piecemeal fashion. If there is consensus that this should be three PRs instead I'm happy to close and re-open, but I also didn't want to spam too much.

The patches are:

* Unify comments, most in shape but to some degree also placement and language
* Use string duplication helper in all places where suitable
* Return CURLcode rather than int with magic numbers in ouput file writing

### CODE REVIEW

File: include/curl/curl.h [hunk 1]
```diff
@@ -612,6 +612,7 @@ typedef enum {
   CURLE_HTTP3,                   /* 95 - An HTTP/3 layer problem */
   CURLE_QUIC_CONNECT_ERROR,      /* 96 - QUIC connection error */
   CURLE_PROXY,                   /* 97 - proxy handshake error */
+  CURLE_OUTPUT_FILE_COULDNT_WRITE_FILE, /* 98 */
```
Comment (jay) [inline include/curl/curl.h:615]: why can't you use curl write error (26) instead

Comment (danielgustafsson) [reply include/curl/curl.h:615]: I keep thinking that's for downloaded data, but I agree you have a good point. Will change to that.

**Comment by bagder** [comment]
And azure goes bananas on test 433... :cry: #6739

**Comment by bagder** [comment]
@danielgustafsson can we get a refreshed/rebased version of this as we approach merge time?

**Comment by danielgustafsson** [comment]
> @danielgustafsson can we get a refreshed/rebased version of this as we approach merge time?

Absolutely, will fix tomorrow.

**Comment by bagder** [comment]
@danielgustafsson would you like me to grab this and take it forward a notch?

**Pull request closed without merging**